packages/plugin-mocks/src/global-types.ts

```diff
@@ -10,6 +10,19 @@ declare global {
 
     export interface BuildSchemaOptions<Types extends SchemaTypes> {
       mocks?: ResolverMap<Types>;
+      typeMocks?: Mock<Types>[];
+    }
+
+    export interface SchemaBuilder<Types extends SchemaTypes> {
+      createObjectMock: <
+        Shape extends NameOrRef extends ObjectParam<Types>
+          ? ShapeFromTypeParam<Types, NameOrRef, false>
+          : object,
+        NameOrRef extends ObjectParam<Types> | string,
+      >(
+        nameOrRef: NameOrRef,
+        resolver: Resolver<unknown, unknown, Types['Context'], Partial<Shape>>,
```

Comment:
what if you have a `resolveList` third argument that defaults to `(...args) => [resolver(...args)]` (just returning a list of 1 item by default, but lets you customize list behavior if you want)